tokio-trace-tower-http/src/lib.rs

```diff
@@ -104,10 +100,18 @@ where
     type Item = InstrumentedHttpService<T::Item>;
     type Error = T::Error;
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-        let span = self.span.clone();
-        self.inner
-            .poll()
-            .map(|ready| ready.map(|svc| InstrumentedHttpService::with_span(svc, span)))
+        match self.inner.poll()? {
+            Async::Ready(svc) => {
+                let svc = InstrumentedHttpService::with_span(
+                    svc,
+                    self.span
+                        .take()
```

Comment:
FWIW, I think a single clone here _would_ be fine, but 👍 for completely eliminating it!